tools/cli_tests/README.md

```diff
@@ -0,0 +1,332 @@
+```
+
+### Setting default resources and parameters for the test VM
+
+By default, the default VM is a `noble` VM with `6G` disk, `1G` memory and `2` cores. All of these parameters can be overridden by specifying `--vm-<option>` parameters listed below.
+
+- `--vm-cpus`: CPU core count
+- `--vm-memory`: Memory size
+- `--vm-disk`: Disk size
+- `--vm-image`: Image
+
+### Timeout and retry attempt counts for commands
+
+All commands have certain timeout values and retry attempts associated with them to provide a stable testing experience. This is to not fail when a certain command may fail a couple of times before succeeding (e.g. `launch` may fail while the daemon is starting), and it's considered fine. While the test code preemptively and defensively tries to remedy these, the `timeout` and `retry` attempts are still useful as a safety net.
+
+#### Timeout values and `--cmd-timeout`
+
+The `timeout` is how long to wait for a certain command to complete. This is enforced at process level and independent from the `--timeout` value the `multipass <command>` has. The timeout values are set on per command basis and each command can have a different timeout value (e.g., `600s` for launch, `60s`for `stop`.) The timeout is applied to per command invocation, so if there are `retries` each retry would get the timeout independently.
+
+The defaults are listed below:
+
+| Command              | Default timeout value |
+| -------------------- | --------------------- |
+| launch               | 600s                  |
+| stop                 | 180s                  |
+| mount                | 180s                  |
+| restart              | 120s                  |
+| delete               | 90s                   |
+| exec                 | 90s                   |
+| start                | 90s                   |
+| umount               | 45s                   |
+| <all other commands> | 30s                   |
+
+`--cmd-timeouts` parameter can be used to change the defaults as follows:
+
+```bash
+pytest tools/cli_tests/ --cmd-timeouts launch=300 start=180 alias=10
+```
+
+#### Retry attempts and `--cmd-retries`
+
+The `retry` is how many times a certain command is going to be `retried` on failure until it's considered a failure. This is to prevent things like `launch` sometimes fails on first invocation because the daemon is not ready yet.
+
+Right now, only the `launch` command has `3` retry attempts by default, while all other commands has `0`.
+
+`--cmd-retries` parameter can be used to change the default as follows:
+
+``` bash
+pytest tools/cli_tests/ --cmd-retries launch=5 exec=2
+```
+
+### Remarks
+
+It's possible to do plethora of things with `pytest` flags. Refer to the [pytest's documentation](https://docs.pytest.org/en/6.2.x/usage.html) to learn more about how to meet your particular needs.
+
+## Running CLI Tests in GitHub Actions
```

Comment:
I feel like everything past this point is a bit verbose and kind of goes without saying. 🤷

Reply:
Do you think it's not worth documenting how the workflow is supposed to be used?

Reply:
Alright, I concede 😄